content/en/platform/corda/5.0/developing-applications/basic-cordapp/flows.md

```diff
@@ -0,0 +1,768 @@
+3. Use `addOutputState` to add the `newStamp` (the created `AppleStamp`).
+4. Use `addCommand` to add the `Issue` command of the `AppleStampContract`.
+5. Use `addSignatories` to add the list of required signatories of the transaction. This should include the `issuer` and the `holder`.
+6. Use `setTimeWindowUntil` to set a time window for the transaction in which the transaction is valid.
+In Corda 5, all transactions using the UTXO ledger must specify an upper bound time window in this way.
+7. Use `toSignedTransaction` to sign the transaction.
+
+This is what your transaction creation code should look like now:
+
+```kotlin
+val newStamp = AppleStamp(
+    id = UUID.randomUUID(),
+    stampDesc = stampDescription,
+    issuer = issuer,
+    holder = holder,
+    participants = listOf(issuer, holder)
+)
+
+val transaction = utxoLedgerService.createTransactionBuilder()
+    .setNotary(notaryInfo.name)
+    .addOutputState(newStamp)
+    .addCommand(AppleCommands.Issue())
+    .setTimeWindowUntil(Instant.now().plus(1, ChronoUnit.DAYS))
+    .addSignatories(listOf(issuer, holder))
+    .toSignedTransaction()
+
+```
+
+#### Finalize the Transaction
+
+Finalizing a transaction does the following:
+
+1. Sends it to counterparties to:
+   * verify the transaction’s contracts.
+   * validate the transaction’s contents.
+   * sign the transaction.
+2. Notarizes the transaction.
+3. Records the transaction for the current participant.
+4. Sends the counterparties the finalized transaction signatures.
+5. The counterparties record the transaction.
+
+To finalize a transaction:
+
+1. Start a `FlowSession` with the `holder` using `FlowMessaging.initiateFlow`:
+   ```kotlin
+   val session = flowMessaging.initiateFlow(holderName)
+   ```
+2. Call `UtxoLedgerService.finalize` and pass the transaction and session in. Include a `try/catch` block around the call:
+   ```kotlin
+   try {
+    // Send the transaction and state to the counterparty and let them sign it
+    // Then notarise and record the transaction in both parties' vaults.
+    utxoLedgerService.finalize(transaction, listOf(session))
+    } catch (e: Exception) {
+    }
+   ```
+
+#### Return a Result from the Flow
+
+The transaction will now be successfully finalized and the end of the flow has been reached.
+The flow must return a `String` representation of the result of the flow.
+Extract and return the `id` of the `newStamp` created earlier (this will be useful in later flows) and return it from the flow. A `String` should also be returned to represent failures occurring inside the finalize call.
+
+This should look like:
+
+```kotlin
+return try {
+    // Send the transaction and state to the counterparty and let them sign it
+    // Then notarise and record the transaction in both parties' vaults.
+    utxoLedgerService.finalize(transaction, listOf(session))
+    newStamp.id.toString()
+} catch (e: Exception) {
+    "Flow failed, message: ${e.message}"
+}
```

Comment:
I think this should be given earlier instead of having the initial explanation in section (2) and then fixing it

Reply:
leaving as is